eos/maths/legendre-polynomial-vector.hh

```diff
@@ -0,0 +1,266 @@
+            {
+                std::vector<double> zerosp = boost::math::legendre_p_zeros<double>(order_);
+                std::array<double, order_/2 + order_ % 2> weightsp;
+                const unsigned len = order_/2 + order_ % 2;
+
+                for (unsigned i = 0; i < len; i++)
+                {
+                    weightsp[i] = 2*(1 - zerosp[i]*zerosp[i])/((order_ + 1)*(order_ + 1)*boost::math::legendre_p<double>(order_ + 1, zerosp[i])*boost::math::legendre_p<double>(order_ + 1, zerosp[i]));
+                }
+
+                // Flip order for first half
+                for (unsigned i = 0; i < len; i++)
+                {
+                    zeros[i] = -1*zerosp[len - 1 - i];
+                    weights[i] = weightsp[len - 1 - i];
+                }
+
+                // Double them, take care in case of odd order_
+                if (order_ % 2 == 1)
+                {
+                    for (unsigned i = 1; i < len; i++)
+                    {
+                        zeros[len + i - 1] = zerosp[i];
+                        weights[len + i - 1] = weightsp[i];
+                    }
+                }
+                else
+                {
+                    for (unsigned i = 0; i < len; i++)
+                    {
+                        zeros[len + i] = zerosp[i];
+                        weights[len + i] = weightsp[i];
+                    }
+                }
+            }
+    };
+
+    /*
+     * Representation of a vector of the real part of the associated Legendre functions for m = 0.
+     * Implementation is based on Zhang, Shanjie and Jin, Jianming. “Computation of Special Functions”, John Wiley and Sons, 1996.
+     *
+     */
+    template <unsigned order_>
+    class LegendreReQVector
+    {
+        private:
+            const double _eps;
+            const unsigned _cut;
+
+        public:
+            LegendreReQVector() :
+                _eps(1e-14),
+                _cut(1000)
+            {
+            }
+
+            LegendreReQVector(const double & eps, const unsigned & cut) :
+                _eps(eps),
+                _cut(cut)
+            {
+            }
+
+            LegendreReQVector(const LegendreReQVector &) = default;
+            LegendreReQVector(LegendreReQVector &&) = default;
+            ~LegendreReQVector() = default;
+            LegendreReQVector & operator= (LegendreReQVector &&) = default;
+            LegendreReQVector & operator= (const LegendreReQVector &) = default;
+
+            // Evaluate the vector of associated Legendre functions
+            // Throws exception if algorithm doesn't terminate
+            // Returns real part if abs(z) > 1
+            std::array<double, order_ + 1> constexpr operator() (const double & z) const
+            {
+                std::array<double, order_ + 1> retVec;
+
+                // Standard forward recursion
+                if (abs(z) < 1)
```

Comment:
In general, we should use ``std::abs()``, ``std::log``, and similar. In a ``.cc`` file, this can be achieved at the top by ``using std::abs`` for the whole translation unit. However, this class template is in a header file, hence we need to use ``std::abs`` everywhere.